modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/restclients/UserStoreMgtRestClient.java

```diff
@@ -0,0 +1,136 @@
+import org.wso2.carbon.identity.user.store.configuration.stub.dto.UserStoreDTO;
+import org.wso2.identity.integration.test.rest.api.server.user.store.v1.model.UserStoreReq;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+public class UserStoreMgtRestClient extends RestBaseClient {
+    private static final String TENANT_PATH = "t/%s";
+    private static final String API_SERVER_BASE_PATH = "/api/server/v1";
+    public static final String USER_STORES_ENDPOINT_URI = "/userstores";
+    private final CloseableHttpClient client;
+    private final String username;
+    private final String password;
+    private final String userStoreBasePath;
+
+    public UserStoreMgtRestClient(String backendURL, Tenant tenantInfo) {
+        client = HttpClients.createDefault();
+
+        this.username = tenantInfo.getContextUser().getUserName();
+        this.password = tenantInfo.getContextUser().getPassword();
+        String tenantDomain = tenantInfo.getContextUser().getUserDomain();
+
+        userStoreBasePath = backendURL + String.format(TENANT_PATH, tenantDomain) + API_SERVER_BASE_PATH +
+                USER_STORES_ENDPOINT_URI;
+    }
+
+    /**
+     * Add a secondary user store.
+     *
+     * @param UserStoreReq Secondary user store request object.
+     */
+    public String addUserStore(UserStoreReq UserStoreReq) throws Exception {
+        String jsonRequest = toJSONString(UserStoreReq);
+        try (CloseableHttpResponse response = getResponseOfHttpPost(userStoreBasePath, jsonRequest, getHeaders())) {
+            String[] locationElements = response.getHeaders(LOCATION_HEADER)[0].toString().split(PATH_SEPARATOR);
+            return locationElements[locationElements.length - 1];
+        }
+    }
+
+    /**
+     * Get secondary user stores.
+     *
+     * @return JSONArray element of the user stores.
+     */
+    public JSONArray getUserStores() throws Exception {
+
+        try (CloseableHttpResponse response = getResponseOfHttpGet(userStoreBasePath, getHeaders())) {
+            return getJSONArray(EntityUtils.toString(response.getEntity()));
+        }
+    }
+
+    /**
+     * Delete a user store
+     *
+     * @param domain User store domain(id).
+     */
+    public void deleteUserStore(String domain) throws IOException {
+        String endpointUrl = userStoreBasePath + PATH_SEPARATOR + domain;
+
+        try (CloseableHttpResponse response = getResponseOfHttpDelete(endpointUrl, getHeaders())) {
+            Assert.assertEquals(response.getStatusLine().getStatusCode(), HttpServletResponse.SC_NO_CONTENT,
+                    "Application deletion failed");
+        }
+    }
+
+    /**
+     * Check user store deployment
+     *
+     * @param domain User Store name
+     * @return boolean response of the user store deployment.
+     */
+    public boolean waitForUserStoreDeployment(String domain) throws Exception {
+
+        long waitTime = System.currentTimeMillis() + 30000; //wait for 45 seconds
```

Comment:
Comment says you want to wait 45 seconds. But you are waiting 30 seconds only.

Reply:
Fixed the comment with https://github.com/wso2/product-is/pull/16354/commits/2e2720d50618a38ef60a3d96e0fe0a7dd5439f26